lib/fs.js

```diff
@@ -827,117 +808,192 @@ fs.ftruncate = function(fd, len, callback) {
   if (typeof len === 'function') {
     callback = len;
     len = 0;
-  } else if (len === undefined) {
+  } else if (len == null) {
```

Comment:
Using the explicit version is actually better performance wise. But do we really want to support using null at all? I did not check but I guess you just ported that over from the c++ layer?

Reply:
yeah, this simplifies an additional check that was being done in C. There's an existing comment in the C code about the inconsistency here.

Reply:
I think == null is optimized anyway

Reply:
@benjamingr that is currently not correct.

Reply:
See #15178.